tools/adventure-pack/src/iteratorPrototypeFilter.ts

```diff
@@ -0,0 +1,23 @@
+import { iteratorPrototype } from "./iteratorPrototype";
+import { iteratorToIterable } from "./iteratorToIterable";
+
+declare global {
+  interface Iterator<T> {
+    filter(
+      callbackFn: (element: T, index: number) => boolean,
+    ): Generator<T, void, undefined>;
+  }
+}
+
+iteratorPrototype.filter = function* <T>(
```

Comment:
If the built-in function already exists (as it will, in a future release), let's not replace it. (I made this update for `.map`)

```suggestion
iteratorPrototype.filter ??= function* <T>(
```